BUILD.rst

```diff
@@ -0,0 +1,195 @@
+=====================================
+Cask Data Application Platform - CDAP
+=====================================
+
+**Standalone and Distributed CDAP**
+
+Building CDAP Maven
+===================
+
+Clean all modules
+-----------------
+
+  mvn clean
+
+Run all tests, fail at the end
+------------------------------
+
+  MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn test -fae
+    
+Build all modules
+-----------------
+
+  mvn package
+
+Run checkstyle, skipping tests
+------------------------------
+
+  mvn package -DskipTests
+
+Build a particular module
+-------------------------
+
+  mvn package -pl [module] -am
+
+Run selected test
+-----------------
+
+  MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn -Dtest=TestClass,TestMore*Class,TestClassMethod#methodName -DfailIfNoTests=false test
+
+See `Surefire doc <http://maven.apache.org/surefire/maven-surefire-plugin/examples/single-test.html`__ for more details
+
+Build all examples
+------------------
+
+  MAVEN_OPTS="-Xmx512m" mvn package -DskipTests -pl cdap-examples -am -amd -P examples
+
+Build Standalone distribution ZIP
+---------------------------------
+
+  MAVEN_OPTS="-Xmx512m" mvn clean package -DskipTests -P examples -pl cdap-examples -am -amd && mvn package -pl cdap-standalone -am -DskipTests -P dist,release
+    
+Build the limited set of Javadocs used in distribution ZIP
+----------------------------------------------------------
+
+  mvn clean package javadoc:javadoc -pl cdap-api -am -DskipTests -P release
+
+Build the complete set of Javadocs, for all modules
+---------------------------------------------------
+
+  mvn clean site -DskipTests
+    
+Build distributions (rpm, deb, tgz)
+-----------------------------------
+
+  mvn package -DskipTests -P dist,rpm-prepare,rpm,deb-prepare,deb,tgz
+
+Show dependency tree
+--------------------
+
+  mvn package dependency:tree -DskipTests
+
+Show dependency tree for a particular module
+--------------------------------------------
+
+  mvn package dependency:tree -DskipTests -pl [module] -am
+
+Show test output to stdout
+--------------------------
+
+  mvn -Dsurefire.redirectTestOutputToFile=false ...
+
+Offline mode
+------------
+
+  mvn -o ....
+
+Change version
+--------------
+
+  mvn versions:set -DnewVersion=[new_version] -DgenerateBackupPoms=false -P examples
+
+
+Description of Modules
```

Comment:
I am not sure why we add this list of module descriptions here, instead of having a Readme in every module. The list here is incomplete, and some information is out of context...